### PR TITLE
 Prepare for XGBoost 2.0

### PR DESCRIPTION
**Pull request by celestinoxp**
Closes #3672

### CODE REVIEW

**Comment by celestinoxp** [comment]
@Yard1 @tvdboom @ngupta23
can anyone check and fix errors?

**Comment by ngupta23** [comment]
You can ignore the python 3.7 errors. We will likely drop support for it in the next release.
Can you please fix the Code Quality issues?

**Comment by celestinoxp** [comment]
@ngupta23 @Yard1 @tvdboom 
tests failing, can you lend a hand?

**Comment by celestinoxp** [comment]
@trivialfis what is need to support xgboost 2.0, change "tree_hist" to "hist" and introduce "device" parameter? can tou check and make any suggestions?

**Review by trivialfis — COMMENTED** [review]
Looks good to me. In addition to the `device` parameter, we have removed the `ntree_limit` attribute, which has been deprecated in the pass.

File: pycaret/containers/models/regression.py [hunk 1]
```diff
@@ -1520,13 +1520,23 @@ def __init__(self, experiment):
...
             "random_state": experiment.seed,
             "n_jobs": experiment.n_jobs_param,
             "verbosity": 0,
             "booster": "gbtree",
-            "tree_method": "gpu_hist" if experiment.gpu_param else "auto",
         }
+
+        if xgboost_2_or_higher:
+            args["tree_method"] = ("hist",)
+            args["device"] = ("cuda",)
```
Comment (Yard1) [inline pycaret/containers/models/regression.py:1536]: are those supposed to be tuples or strings?

File: pycaret/containers/models/regression.py [hunk 2]
```diff
@@ -1520,13 +1520,23 @@ def __init__(self, experiment):
 
         from xgboost import XGBRegressor
 
+        # If XGBoost >= 2 change and add new parameters
+        xgboost_version = tuple(map(int, xgboost.__version__.split(".")))
```
Comment (Yard1) [inline pycaret/containers/models/regression.py:1524]: Let's use `packaging.version.parse` instead.

**Review by Yard1 — CHANGES_REQUESTED** [review]
Thanks, 2 comments

File: pycaret/containers/models/regression.py (hunk 1 above)
Comment (celestinoxp) [reply pycaret/containers/models/regression.py]: I forgot it doesn't need commas, so black automatically formatted it as tuples. adjusted. thanks

File: pycaret/containers/models/regression.py (hunk 2 above)
Comment (celestinoxp) [reply pycaret/containers/models/regression.py]: fixed. thanks

**Comment by ngupta23** [comment]
@celestinoxp - Your tests are failing due to these 3 tests. Can you run them locally to reproduce and fix the issue?

image

**Comment by celestinoxp** [comment]
@ngupta23
i think it is a numpy problem... can you help? (Allow edits by maintainers is activated :) )

**Pull request closed without merging**